### PR TITLE
18535 - Save and Resume Draft Amalgamation

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#18535

*Description of changes:*
- Save and resume draft Amalgamation Filing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-592-6275tvzb.web.app

File: src/components/Amalgamation/AmalgamatingBusinesses.vue [hunk 1]
```diff
@@ -176,6 +176,11 @@ export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
 
   readonly BusinessLookupServices = BusinessLookupServices
 
+  // If continuing a draft, initialize the amalgamatingBusinesses array as the previously saved one.
+  mounted (): void {
+    if (this.getAmalgamatingBusinesses) this.amalgamatingBusinesses = this.getAmalgamatingBusinesses
+  }
```
Comment (severinbeauvais) [inline src/components/Amalgamation/AmalgamatingBusinesses.vue:182]: OK, so you have the array in the store and a local copy here... I may rework this after you merge and I rebase.

File: src/store/store.ts [hunk 2]
```diff
@@ -1232,6 +1237,9 @@ export const useStore = defineStore('store', {
     setAmalgamatingBusinessesValid (valid: boolean) {
       this.stateModel.amalgamation.amalgamatingBusinessesValid = valid
     },
+    setAmalgamationCourtApproval (courtApproval: boolean) {
+      this.stateModel.amalgamation.courtApproval = courtApproval
+    },
```
Comment (severinbeauvais) [inline src/store/store.ts:1242]: I thought we had a ticket to implement the court approval component, but I can't find it. Can you? If not, please create one in the epic that we can groom tomorrow.

Comment (JazzarKarim) [reply src/store/store.ts:1242]: Isn't it this one Sev? https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/18749

Comment (severinbeauvais) [reply src/store/store.ts:1242]: Yes, thanks. (I've updated its title to make it easier to find next time 😉 )

I guess you can check off a couple of items in that ticket.

Comment (JazzarKarim) [reply src/store/store.ts:1242]: Sounds good, will do that.

**Comment by JazzarKarim** [comment]
> > **UPDATE:** After Kevin merged his PR, I'm able to save the draft and load it properly!
> 
> That's great! Does it also resume correctly?
> 
> I see we store the entire table (not the minimum specified by the schema). I think the Legal API won't care, but we need to be sure that each business is completely re-fetched when a draft is resumed. Does that fit in your existing ticket or do you want to do some of that as part of 18641?

Yup, it does! I've tried adding and saving + deleting and saving, works well.

Let's keep that as a part of [#18641](https://github.com/bcgov/entity/issues/18641). I'll also mention that there.

**Pull request merged**